lib/build/initializer_plugin.dart

```diff
@@ -273,15 +274,24 @@ class DefaultInitializerPlugin implements InitializerPlugin {
           'And got the following errors: ${result.errors}.');
       return null;
     }
-    var value = result.value.value;
+
+    var value = result.value.toStringValue();
+    if (value != null) value = _stringValue(value);
+
+    if (value == null) value = result.value.toBoolValue();
+    if (value == null) value = result.value.toIntValue();
+    if (value == null) value = result.value.toDoubleValue();
+    if (value == null) value = result.value.toListValue();
+    if (value == null) value = result.value.toMapValue();
+    if (value == null) value = result.value.toSymbolValue();
+    if (value == null) value = result.value.toTypeValue();
```

Comment:
Ok, I will steal that code from source_gen. I definitely think this utility should be provided as part of the analyzer as its pretty common.